lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake

```diff
@@ -0,0 +1,32 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.8' do
+      desc "Clear checksum type for on-demand repositories"
+      task :clear_checksum_type => %w(environment) do
+        User.current = User.anonymous_admin
+
+        repos = Katello::Repository.yum_type.where.not(checksum_type: nil, url: nil)
```

Comment:
Testing the upgrade scenario, i still got failures.  I think this should just be `Katello::Repository.yum_type`

and the 'if repo.on_demand?' below should be 'if repo.on_demand? && !repo.url.blank?`